### PR TITLE
Render tests

### PR DESCRIPTION
**Pull request by recrit**

### CODE REVIEW

File: src/Property/Component/AbstractComponent.php [hunk 1]
```diff
@@ -32,6 +31,7 @@
 
     /**
      * Twig environmnt object.
```
Comment (andrewmriley) [inline src/Property/Component/AbstractComponent.php]: Typo in "environment"

File: tests/AbstractTest.php [hunk 2]
```diff
@@ -5,33 +5,39 @@
 abstract class AbstractTest extends \PHPUnit_Framework_TestCase
 {
     /**
+     * Twig environmnt object.
```
Comment (andrewmriley) [inline tests/AbstractTest.php]: Typo in environment

File: tests/AbstractTest.php [hunk 3]
```diff
@@ -5,33 +5,39 @@
...
      * Provides a mocked Configuration object.
      */
     public function getConfig()
     {
-        $logger = $this->getMockBuilder('Psr\Log\LoggerInterface')
-            ->getMock();
+        $twig = $this->getTwig();
+        $retriever = new \JsonSchema\Uri\UriRetriever();
+        $logger = new \Psr\Log\NullLogger();
+        $retriever = new \JsonSchema\Uri\UriRetriever();
```
Comment (andrewmriley) [inline tests/AbstractTest.php:23]: I don't see retriever being used.

Comment (recrit) [inline tests/AbstractTest.php:23]: It gets passed into the resolver on the next line.

`$resolver = new \JsonSchema\RefResolver($retriever);`

Comment (recrit) [inline tests/AbstractTest.php:23]: The 1st one is a duplicate... removing

File: tests/AbstractTest.php [hunk 4]
```diff
@@ -41,11 +47,21 @@ public function getConfig()
      */
```
Comment (andrewmriley) [inline tests/AbstractTest.php:52]: There should be a return line

File: tests/AbstractTest.php [hunk 5]
```diff
@@ -60,4 +76,45 @@ public function getJson($filename)
             return file_get_contents($filepath);
         }
     }
+
+    /**
+     * Helper function to set flat data on a component.
+     *
+     * @param \PatternBuilder\Property\PropertyInterface $property
```
Comment (andrewmriley) [inline tests/AbstractTest.php]: Param is incorrect

File: tests/AbstractTest.php [hunk 6]
```diff
@@ -60,4 +76,45 @@ public function getJson($filename)
...
+     *                                                             The property / component object.
+     * @param array                                      $values
+     *                                                             An array with keys of omponent property name and values to set.
```
Comment (andrewmriley) [inline tests/AbstractTest.php]: Missing the C in component

File: tests/RenderTest.php [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        // Check assert values.
+        $this->checkMarkupContainsValues($markup, $assert_values);
+    }
+
+    /**
+     * Render tests for schema references.
+     */
+    public function testSchemaReference()
+    {
+        $twig = $this->getTwig();
```
Comment (andrewmriley) [inline tests/RenderTest.php]: $twig isn't used.

**Comment by andrewmriley** [comment]
Code review looks good

**Pull request merged**